networkx/classes/function.py

```diff
@@ -288,33 +292,51 @@ def info(G, n=None):
 
 
 def set_node_attributes(G, name, values):
-    """Set node attributes from dictionary of nodes and values
+    """Sets node attributes from a given value or dictionary of values.
 
     Parameters
     ----------
     G : NetworkX Graph
 
     name : string
-       Attribute name
+       Name of the node attribute to set.
 
-    values: dict
-       Dictionary of attribute values keyed by node. If `values` is not a
-       dictionary, then it is treated as a single attribute value that is then
-       applied to every node in `G`.
+    values : dict
+       Dictionary of attribute values keyed by node. If ``values`` is
```

Comment:
OK... I see it in #1423 

Some investigating shows that the trouble is that we have configured sphinx in ```doc/source/conf.py``` with a line ```default_role = 'math'```. This is supposed to make math mode easy. But it doesn't follow the standard numpydoc role ```:obj:``` for back-ticks. The ```:obj:``` role makes a link to the docs for the python object with that name, or emphasizes it if no such object is found.

I think we should change the default_role from ```:math:``` to ```:obj:```.   That might affect a lot of docstrings.  I'll open another issue for it.